taskvine/src/manager/vine_schedule.c

```diff
@@ -145,26 +145,36 @@ static struct vine_worker_info *find_worker_by_files(struct vine_manager *q, str
 	struct vine_worker_info *best_worker = 0;
 	int64_t most_task_cached_bytes = 0;
 	int64_t task_cached_bytes;
+	uint8_t has_all_files;
 	struct vine_file_replica *remote_info;
 	struct vine_mount *m;
 
-	HASH_TABLE_ITERATE(q->worker_table,key,w) {
+	HASH_TABLE_ITERATE(q->worker_table, key, w) {
 		if( check_worker_against_task(q, w, t) ) {
 			task_cached_bytes = 0;
+			has_all_files = 1;
+
 			LIST_ITERATE(t->input_mounts,m) {
```

Comment:
@colinthomas-z80 Maybe you should also skip over files that have VINE_NO_CACHE. If a task has no inputs marked as cached, then any worker would do.